### ISSUE TITLE
[Bug]: Breaking change introduced with 4.6.3 release for defaultValue on nested paths using flattened keys

### ISSUE THREAD

**Issue by ZipBrandon** [airjp73/remix-validated-form#204]
### Which packages are impacted?

- [X] `remix-validated-form`
- [ ] `@remix-validated-form/with-zod`
- [ ] `@remix-validated-form/with-yup`
- [ ] `zod-form-data`

### What version of these packages are you using?

 "remix-validated-form": "4.6.3"

### Please provide a link to a minimal reproduction of the issue.

https://github.com/airjp73/remix-validated-form/issues

### Steps to Reproduce the Bug or Issue

It seems that setting default values for nested paths no longer works. Default values now come in as undefined when utilizing an object with flattened keys like: 
```
{
'foo.bar': 'baz'
}
```

It does work as expected 
```
{
 foo: {
bar: 'baz'
 }
}
```

### Expected behavior

Default values to continue to be set like they were in 4.6.2 and previous, or at least an announcement if it is intended.

### Platform

All

**Comment by airjp73** [comment]
Hi!

This is interesting. This wasn't an unintended change, but it was also an unintended behavior. I'm kinda surprised that ever worked to begin with. 😅

Seems like a cool feature though, so I'm open to supporting this.

I recently swapped out the Lodash `get` and `set` functions for a custom implementation in order to decrease bundle size -- there's likely a difference in behavior and we just need to update one or both of them.

A couple open questions:

What's the expected behavior in a case like this?

```
{
  foo: { bar: 'one' },
  'foo.bar': 'two',
}
```

My instinct here is the throw rather than relying on key order.

And what about this case?

```
{
  foo: {
    'bar.baz': 123
  }
}
```

Here I think they'd probably build on each other to result in 'foo.bar.baz'.

**Comment by ZipBrandon** [comment]
Awesome. My preference is also to throw on the first.  I would expect the second to be equivalent to both `{"foo.bar.baz": 123}` and `{foo:{bar:{baz: 123}}}`

Lodash's `get/set` behavior ensured that pathing specifications worked.  I'm currently pinned to `4.6.2` but looking forward to come back to `latest`!  Thank you!

**Comment by airjp73** [comment]
Fix released in 4.6.7
